services/vision/obstaclesdepth/obstacles_depth.go

```diff
@@ -0,0 +1,329 @@
+// Package obstaclesdepth uses an underlying depth camera to fulfill GetObjectPointClouds,
+// using the method outlined in (Manduchi, Roberto, et al. "Obstacle detection and terrain classification
+// for autonomous off-road navigation." Autonomous robots 18 (2005): 81-102.)
+package obstaclesdepth
+
+import (
+	"context"
+	"image"
+	"math"
+	"strconv"
+
+	"github.com/edaniels/golog"
+	"github.com/golang/geo/r3"
+	"github.com/muesli/clusters"
+	"github.com/muesli/kmeans"
+	"github.com/pkg/errors"
+	"github.com/viamrobotics/gostream"
+	"go.opencensus.io/trace"
+	goutils "go.viam.com/utils"
+
+	"go.viam.com/rdk/components/camera"
+	"go.viam.com/rdk/pointcloud"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/rimage"
+	"go.viam.com/rdk/rimage/transform"
+	"go.viam.com/rdk/robot"
+	svision "go.viam.com/rdk/services/vision"
+	"go.viam.com/rdk/spatialmath"
+	"go.viam.com/rdk/utils"
+	vision "go.viam.com/rdk/vision"
+	"go.viam.com/rdk/vision/segmentation"
+)
+
+var model = resource.DefaultModelFamily.WithModel("obstacles_depth")
+
+// ObsDepthConfig specifies the parameters to be used for the obstacle depth service.
+type ObsDepthConfig struct {
+	Hmin       float64 `json:"h_min_m"`
+	Hmax       float64 `json:"h_max_m"`
+	ThetaMax   float64 `json:"theta_max_deg"`
+	ReturnPCDs bool    `json:"return_pcds"`
+	CameraName string  `json:"camera_name"`
```

Comment:
just a question - are you still planning on doing the map of all camera intrinsics, or switching to the standard component constructor and using the dependencies to get the camera?